### ISSUE TITLE
"GQRX cannot be opened because of a problem"

### ISSUE THREAD

**Issue by davidcranor** [chleggett/homebrew-gqrx#8]
I have success building and linking to my Application folder, but when I try to click on the icon it errors out and says "GQRX cannot be opened because of a problem."

I'm on OSX 10.10.5.  Any ideas as to what could be the issue?

Thanks!
David
<img width="542" alt="screen shot 2015-09-21 at 2 19 52 am" src="https://cloud.githubusercontent.com/assets/373589/9986188/4b101f46-6007-11e5-8881-645ee654040f.png">

**Comment by chleggett** [comment]
It's been a while, so you may have already figured this out, but incase not ...

I see a different issue in the screenshot that you posted than the "when I try to click on the icon it errors out" that you mention.

I'll try to address both. Even if it doesn't help you, it may help someone else.

### "gqrx: command not found" in your screenshot

This formula compiles and creates a Gqrx application which is a GUI application like Safari, etc. not one that you use on the command line.

Look in your /Applications folder, and you should see /Applications/Gqrx.app you can double click it to run it.

When it is running, drag it to wherever you want it to live on your Dock for easy access next time.

### "when I try to click on the icon it errors out and says "GQRX cannot be opened because of a problem.""
 
Coming back full circle, you actually can launch it from the command line in two ways:
 - open /Applications/Gqrx.app
 - open /Applications/Gqrx.app/Contents/MacOS/Gqrx

The first one will launch it and give you your command line back.

The second one will spawn another terminal window with debugging information.

Do the second one and cut and paste the results.

**Comment by davidcranor** [comment]
Hey!  No, I did not figure it out.  Appended is the output from command #2.  seems like it is unhappy about the presence of my libgnuradio.  I've brew installed gnuradio without complaint from metacollin's keg, though....

**********************

> /usr/local/Cellar/gqrx/2.3.2/Gqrx.app/Contents/MacOS/Gqrx ; exit 
dyld: Library not loaded: /usr/local/lib/libgnuradio-runtime.3.7.7.dylib
  Referenced from: /usr/local/lib/libgnuradio-osmosdr.0.1.4.dylib
  Reason: image not found
Trace/BPT trap: 5
logout

**Comment by chleggett** [comment]
There is a recent development and if you don't mind trying it out, I'd appreciate it.

There is now an "official" formula for gnuradio in the homebrew repository.

Will you please move off of metacollins' formula and over to that one.

Here's the process that I think will work for you (it uses the rmtree package that you may already have):
    $ brew tap beeftornado/rmtree && brew install beeftornado/rmtree/brew-rmtree
    $ brew rmtree metacollin/gnuradio/gnuradio
    $ brew uninstall metacollin/gnuradio/gnuradio
    $ brew untap metacollin/gnuradio
    $ brew install gnuradio
    $ brew install gqrx

Let me know what happens.

**Comment by davidcranor** [comment]
OK, so it runs, but when I try to select my RTL-SDR as an input source, the window dies and respawns.  I can see Realtek RTL in the source list, but as soon as I click OK everything resets.  

***************************
;> /usr/local/Cellar/gqrx/2.3.2/Gqrx.app/Contents/MacOS/Gqrx ; exit 
Mac OS; Clang version 7.0.0 (clang-700.0.72); Boost_105800; UHD_003.009.001-0-unknown

gr-osmosdr 0.1.4 (0.1.4) gnuradio 3.7.8
built-in source types: file fcd rtl rtl_tcp uhd rfspace 
Using Volk machine: avx_64_mmx
INFO: Audio sink arch: osx
INFO: 

Using output audio device 'Built-in Output'.
  ... which is the current default output audio device.
  Changing the default output audio device in the System Preferences will 
  result in changing it here, too (with an internal reconfiguration).

Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
Could not resolve property : pattern10600
gr-osmosdr 0.1.4 (0.1.4) gnuradio 3.7.8
built-in source types: file fcd rtl rtl_tcp uhd rfspace 
Using device #0 Realtek RTL2838UHIDIR SN: 00000001
Found Rafael Micro R820T tuner
[R82XX] PLL not locked!
Exact sample rate is: 1500000.014901 Hz
[R82XX] PLL not locked!
Number of audio input devices: 1
Number of audio output devices: 2
2015-09-30 14:01:02.520 Gqrx[6577:1920605] modalSession has been exited prematurely - check for a reentrant call to endModalSession:
logout

[Process completed]



![screen shot 2015-09-30 at 2 00 50 pm](https://cloud.githubusercontent.com/assets/373589/10201822/ee1dafca-677b-11e5-8fa8-cab72262239b.png)


Thanks for your help on this!

**Comment by davidcranor** [comment]
Argh, now gnuradio-companion won't open.  Did some digging, everyone says to use the macports version of the gnuradio installer?  Argh, I don't want to have it alongside homebrew...

**Comment by natevw** [comment]
I'm having this same trouble, actually brought on by doing a `brew upgrade` for the first time after the gnuradio package is in the main set of Formula.

I tried doing the above steps (rmtree, etc.) except that I used `brew install --build-from-source gnuradio` and `brew install --build-from-source gqrx` to avoid brew installing it's silly own version of Python.

In my case, after making sure I had the old gnuradio cleaned up and the new build installed, nothing changed. I still get:

```
$ /Applications/Gqrx.app/Contents/MacOS/Gqrx
dyld: Library not loaded: /usr/local/lib/libgnuradio-runtime.3.7.7.dylib
  Referenced from: /usr/local/lib/libgnuradio-osmosdr.0.1.4.dylib
  Reason: image not found
Trace/BPT trap: 5
```

Really going to miss Qgrx until I can somehow get it working again :-( :-( :-(

**Comment by natevw** [comment]
Whoohoo, figured this out, at least in my case.

I noticed that `gr-osmosdr` was its own package/formula/dependency/whatever (from this tap apparently). What I did was:

```
brew uninstall gr-osmosdr
brew install --build-from-source gr-osmosdr
```

…and after one crash (`libc++abi.dylib: terminating with uncaught exception of type boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::math::rounding_error> >: Error in function boost::math::round<float>(float): Value inf can not be represented in the target integer type. Abort trap: 6`) it cleared the settings and seems to be working again!
